### PR TITLE
Added type definitions for Notification

### PR DESCRIPTION
**Pull request by singhshashi**
This is for the issue I created https://github.com/facebook/flow/issues/3784. 

Added type definitions based on documentation at https://developer.mozilla.org/en-US/docs/Web/API/Notification

### CODE REVIEW

File: lib/dom.js [hunk 1]
```diff
@@ -3263,3 +3263,25 @@ declare var status: string;
 declare var top: WindowProxy;
 declare function getSelection(): Selection | null;
 declare var customElements: CustomElementRegistry;
+
+/* Notification */
+declare class Notification extends EventTarget {
+  static permission: string;
```
Comment (gabelevi) [inline lib/dom.js]: Maybe 

    type NotificationPermission = 'default' | 'denied' | 'granted';

and then change this to

    static permission: NotificationPermission;

? https://notifications.spec.whatwg.org/#enumdef-notificationpermission

File: lib/dom.js [hunk 2]
```diff
@@ -3263,3 +3263,25 @@ declare var status: string;
 declare var top: WindowProxy;
 declare function getSelection(): Selection | null;
 declare var customElements: CustomElementRegistry;
+
+/* Notification */
+declare class Notification extends EventTarget {
```
Comment (gabelevi) [inline lib/dom.js]: Looking at the spec, I think there are also these:

    static maxActions: number;
    static requestPermission(callback?: (perm: NotificationPermission) => mixed): Promise<NotificationPermission>;
    vibrate: Array<number>;
    renotify: boolean;

https://notifications.spec.whatwg.org/#api

File: lib/dom.js [hunk 3]
```diff
@@ -3263,3 +3263,25 @@ declare var status: string;
...
+  dir: string;
+  lang: string;
+  tag: string;
+  icon: string;
+  image: string;
+  requireInteraction: boolean;
+  silent: boolean;
+  timestamp: number;
+  title: string;
+  onclick: (evt: any) => any;
```
Comment (gabelevi) [inline lib/dom.js]: I think these must take events, right? So maybe

    onclick: (evt: Event) => any;
    onerror: (evt: Event) => any;

https://notifications.spec.whatwg.org/#api

File: lib/dom.js [hunk 4]
```diff
@@ -3263,3 +3263,25 @@ declare var status: string;
...
 declare var customElements: CustomElementRegistry;
+
+/* Notification */
+declare class Notification extends EventTarget {
+  static permission: string;
+  actions: Array<any>;
+  badge: string;
+  body: string;
+  data: string;
+  dir: string;
```
Comment (gabelevi) [inline lib/dom.js]: I think this is

    dir: 'auto' | 'ltr' | 'rtl';

https://notifications.spec.whatwg.org/#enumdef-notificationdirection

File: lib/dom.js [hunk 5]
```diff
@@ -3263,3 +3263,25 @@ declare var status: string;
...
 declare function getSelection(): Selection | null;
 declare var customElements: CustomElementRegistry;
+
+/* Notification */
+declare class Notification extends EventTarget {
+  static permission: string;
+  actions: Array<any>;
+  badge: string;
+  body: string;
+  data: string;
```
Comment (gabelevi) [inline lib/dom.js]: The spec seems to type this as `any`

https://notifications.spec.whatwg.org/#api

File: lib/dom.js [hunk 6]
```diff
@@ -3263,3 +3263,25 @@ declare var status: string;
 declare var top: WindowProxy;
 declare function getSelection(): Selection | null;
 declare var customElements: CustomElementRegistry;
+
+/* Notification */
+declare class Notification extends EventTarget {
+  static permission: string;
+  actions: Array<any>;
```
Comment (gabelevi) [inline lib/dom.js]: I think this is

    actions: Array<{action: string, title: string, icon?: string}>;

**Comment by gabelevi** [comment]
Awesome! Thanks for updating this PR so quick! And thanks for re-ordering the fields to match the spec! It made the second review super easy!

**Comment by singhshashi** [comment]
What else do I need to change for this pull request to get merged? The changes have been approved. I am not sure why the AppVeyor build has failed. Please suggest next steps.

**Comment by dahliasalem** [comment]
Missing constructor declaration @singhshashi

**Comment by singhshashi** [comment]
I do hope this gets merged now.

**Comment by singhshashi** [comment]
@gabelevi Could you please have a look at this again and merge, if everything seems fine? I had made changes as per updated spec and both the checks have passed this time.

**Comment by mrkev** [comment]
I'm gonna need you to rebase it sorry 😅don't worry if the tests fail because of line numbers btw-- I can re-record those.

File: lib/dom.js [hunk 7]
```diff
@@ -3742,3 +3742,54 @@ declare var status: string;
...
+                             timestamp: number, 
+                             renotify: boolean, 
+                             silent: boolean, 
+                             requireInteraction: boolean,
+                             data: ?any, 
+                             actions: Array<NotificationAction>                                                      
+                            }
+
+declare class Notification extends EventTarget {
+  constructor(title: string, options?: NotificationOptions) : void;
```
Comment (pronebird) [inline lib/dom.js]: I think it should be `$Shape<NotificationOptions>`

**Comment by pronebird** [comment]
@mrkev how did rebase go? Could you please have another look at this PR?

**Pull request closed without merging**